lib/chef/provider/package.rb

```diff
@@ -304,6 +305,14 @@ def unlock_package(name, version)
         raise( Chef::Exceptions::UnsupportedAction, "#{self} does not support :unlock" )
       end
 
+      def validate_package(name)
+        raise( Chef::Exceptions::Package, "package name not allowed spaces, tabs, newlines, etc." ) if invalid_name?(name)
+      end
+
+      def invalid_name?(x)
```

Comment:
Thinking about this more, we should change `x` to `name` or `package_name` to be consistent with the rest of the code and make it a little clearer what `x` is evaluating. 

`name.strip.index(/\s/) }` can be reduced to `name =~ /\s/` since we are just checking to see if a value contains white spaces... but we also may want to check for tabs in the regex as mentioned in the exception message.